thunder/tests/opinfos.py

```diff
@@ -6312,6 +6312,75 @@ def sample_generator(op, device, dtype, requires_grad, **kwargs):
 nn_ops.append(avg_pool3d_opinfo)
 
 
+def adaptive_avg_pool2d_error_generator(op, device, **kwargs):
+    make = partial(make_tensor, device=device, dtype=torch.float32)
+
+    cases_runtime_error = (
+        ((3,), (1, 1), "adaptive_avg_pool2d: Expected 3D or 4D tensor, but got"),
+        ((3, 4, 5), (3,), "adaptive_avg_pool2d: output_size must be 2"),
+        (
+            (3, 4, 5),
+            (3, -2),
+            "adaptive_avg_pool2d: elements of output_size must be greater than or equal to 0 but received",
+        ),
+        (
+            (3, 4, 0),
+            (3, 2),
+            "adaptive_avg_pool2d: Expected input to have non-zero size for non-batch dimensions, but input has sizes ",
+        ),
+        (
+            (1, 3, 0, 4),
+            (3, 2),
+            "adaptive_avg_pool2d: Expected input to have non-zero size for non-batch dimensions, but input has sizes ",
+        ),
+    )
+    cases_value_error = (
+        ((3, 4, 5), (3.0, 3.0), r"Element (.*?) \((.*?)\) had an unexpected type"),
+        ((3, 4, 5), 4.0, r"(.*?) had an unexpected type"),
+    )
+    for input_args, err_type in zip((cases_value_error, cases_runtime_error), (ValueError, RuntimeError)):
+        for op_shapes, output_sizes, err_msg in input_args:
+            yield (
+                SampleInput(
+                    make(op_shapes),
+                    output_sizes,
+                ),
+                err_type,
+                err_msg,
+            )
+
+
+def adaptive_avg_pool2d_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    cases = (
+        ((3, 3, 3), 5),
+        ((3, 4, 4), (3, 3)),
```

Comment:
meta seems to accept `output_size` with zeroes? Let's test it?

Reply:
And if it fails in backward, we can short-circuit and return `zeros_like(a)` in the grad transforms.

Reply:
I get it to behave the same as torch cuda, return `zeros_like` when `a.numel=0`, and if grad has zero size non-batch dimension it throws error